### ISSUE TITLE
Developer section

### ISSUE THREAD

**Issue by nicholas-leonard** [nicholas-leonard/torch-doc#3]
Small articles on:
- Writing a lua package for torch
- Writing a C package for torch (torch function or nn module in C)
- Extending a neural network module, and/or writing your own, and unit-testing with nn.Jacobian

**Comment by Atcold** [comment]
Number 2 is already done. See [here](https://github.com/Atcold/Under-the-Hood-of-Torch/blob/master/FFI-so/README.md).
Moreover, shouldn't this repo go under the Torch group rather than be a user repo?

**Comment by nicholas-leonard** [comment]
Yeah but that is specifically the new-school ffi method. We should also include an old-school (current-school) how to build your own nn module in C and CUDA

**Comment by Atcold** [comment]
I personally think that would simply be a waste of time...
There are new easier way. Why would you take the time to explain something no one is going to use?

**Comment by soumith** [comment]
@Atcold fine by me.
